### ISSUE TITLE
Request - make docked panels optional

### ISSUE THREAD

**Issue by olibclarke** [pemsley/coot#84]
Hi,

The option to have validation output (e.g. the Ramachandran plot and validation outliers list) docked to the main window in cootgtk4 is very nice to have, but not always desirable. 

For example, if I am working on a small laptop screen, I would rather have these items as separate windows, and similarly if I am working with multiple displays I would like to be able to arrange these windows as I please. Would it be possible to add an option to undock them from the main window?

Cheers
Oli

**Comment by pemsley** [comment]
It is not clear to me if you know that these are slideable panes. Does that make a difference?

**Comment by olibclarke** [comment]
Yes, I do. One issue with that is that once they are fully slid over, there is no visible sign that they are there, and the mouse target for resizing the window to the left, the mouse target for re-sliding the panel, and the mouse target for rotating the view (all of which are click-drag actions) are extremely close together, e.g.:

https://www.dropbox.com/scl/fi/2q8cidiiay0io4m2t57pp/rama_behavior.mov?rlkey=cv4ezl38d7cqo34rsv9te8oru&dl=0

But the other thing is, I don't want panels obscuring or changing the view - I would rather have panels that I can move around and put beside the thing I'm working on, or potentially on another monitor. Particularly when for example both the Rama plot and the validation list is open, the left side starts feeling a bit cramped IMO - it would be nice at least to have the option of having them as floating windows

**Comment by pemsley** [comment]
@hgonomeg I invite your comments on the above comments.

I think we should add a key-binding to toggle the visibility of vertical validation panel.

**Comment by olibclarke** [comment]
I'm also unconvinced of the utility of slidable panes that allow you to see half a ramachandran plot... if they are going to be slidable I would favor more of a hide/show "drawer" scheme than a continuous slide.

But I really do think they ought to be freely movable/rearrangeable - it adds a lot more flexibility depending how the user works and their display setup. See ChimeraX for a good example of how this can work using the Qt framework - I'm not sure whether there is similar functionality in Gtk4 or not?

**Comment by olibclarke** [comment]
What about having it as a global preference (to have them as panes or resizable floating windows)? Then it doesn't need a dynamic docking/undocking process

**Comment by pemsley** [comment]
Isn't a layout manager what we want? (for rearranging widgets, not docking and undocking)

https://blog.gtk.org/2019/03/27/layout-managers-in-gtk-4/

**Comment by hgonomeg** [comment]
> Isn't a layout manager what we want? (for rearranging widgets, not docking and undocking)
> 
> https://blog.gtk.org/2019/03/27/layout-managers-in-gtk-4/

To my understanding, layout managers are used to govern the position of child widgets inside a single parent widget.
I'm not sure what kind of rearrangement you mean specifically, but I don't think this is what we're looking for here.

**Comment by pemsley** [comment]
Somewhat related video:
https://www.youtube.com/watch?v=BmFMQOKvtEM

**Comment by olibclarke** [comment]
I would favor just reverting it to a multi window app, as in previous versions, and abandon the docked panels for now. It would at least get around current issues with Gtk4 re window focus, and just in general be more flexible. Then users can use their own window manager (e.g. app exposé on Mac, or third party apps like BetterSnapTool) to manage and rearrange individual windows.
